Source/AwsCommonRuntimeKit/crt/ShutdownCallbackCore.swift

```diff
@@ -0,0 +1,83 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import AwsCCommon
+import AwsCAuth
+
+public typealias ShutdownCallback = () -> Void
+
+/// Core classes have manual memory management.
+/// You have to balance the retain & release calls in all cases to avoid leaking memory.
+class ShutdownCallbackCore {
+    let shutdownCallback: ShutdownCallback
+    init?(_ shutdownCallback: ShutdownCallback?) {
```

Comment:
debatable: I wouldn't personally bother with the complexity of making this Core class optional. I'd always have it exist, even if the user didn't pass a shutdownCallback. It's just simpler. Easier to catch reference counting mistakes when you're exercising the code every time...

but if you really think it's worthwhile I won't stop it

Reply:
Thanks, updated.